### PR TITLE
Add radix sort and optimize percentile_approx by using it instead of std::sort (#2102)

### PR DESCRIPTION
**Pull request by GOGOYAO**
RadixSort is a linear time sorting algorithm but std::sort, which is used in Doris `percentile_approx` function, is O(n*log(n)). So we replace the sorting algorithm and optimize the performance for `percentile_approx` function. 

Further more, in TDigest, we use float instead of double to get more improvement with some loss of precision.

Issue #2102

### CODE REVIEW

File: be/test/util/CMakeLists.txt [hunk 1]
```diff
@@ -49,3 +49,4 @@ ADD_BE_TEST(arrow/arrow_row_batch_test)
 ADD_BE_TEST(counter_cond_variable_test)
 ADD_BE_TEST(frame_of_reference_coding_test)
 ADD_BE_TEST(bit_stream_utils_test)
+ADD_BE_TEST(radix_sort_test)
```
Comment (imay) [inline be/test/util/CMakeLists.txt]: should add it to run-ut.sh

File: be/src/util/tdigest.h [hunk 2]
```diff
@@ -612,7 +633,8 @@ class TDigest {
     // when complete, _unprocessed will be empty and _processed will have at most _max_processed centroids
     inline void process() {
         CentroidComparator cc;
-        std::sort(_unprocessed.begin(), _unprocessed.end(), cc);
+        //std::sort(_unprocessed.begin(), _unprocessed.end(), cc);
```
Comment (imay) [inline be/src/util/tdigest.h]: remove this

File: be/src/util/radix_sort.h [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+using RadixSortNumTraits =
+    conditional_t<std::is_integral<T>::value,
+        conditional_t<std::is_unsigned<T>::value,
+            RadixSortUIntTraits<T>,
+            RadixSortIntTraits<T>>,
+        RadixSortFloatTraits<T>>;
+
+
+template <typename Traits>
+struct RadixSort {
```
Comment (imay) [inline be/src/util/radix_sort.h]: should add some comment to describe how to use it.

File: be/test/util/CMakeLists.txt (hunk 1 above)
Comment (shengyunyao) [reply be/test/util/CMakeLists.txt]: done

File: be/src/util/tdigest.h (hunk 2 above)
Comment (shengyunyao) [reply be/src/util/tdigest.h]: done

File: be/src/util/radix_sort.h (hunk 3 above)
Comment (shengyunyao) [reply be/src/util/radix_sort.h]: done

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**